### DISCUSSION TITLE
在低版本linux系统中启动ilogtail报libc版本低的错误

### DISCUSSION THREAD

**Discussion by yanggelinux** [alibaba/ilogtail#500] [Help] [answered]
**在centos6的机器上运行ilogtail 报 libc 版本低的问题，有啥好解决办法吗，编译时候需要加上什么参数吗？**

[root@astable-mysql-1 work]# ./ilogtail
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.14' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.5' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.19' not found (required by ./ilogtail)
./ilogtail: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by ./ilogtail)
./ilogtail: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by ./ilogtail)

**Comment by yyuuttaaoo** [comment, marked as answer]
[core/CMakeList.txt](https://github.com/alibaba/ilogtail/blob/main/core/CMakeLists.txt)中有2个编译开关可以启动低版本兼容模式
option(ENABLE_COMPATIBLE_MODE "Build Logtail in compatible mode (for low version Linux)" OFF)
option(ENABLE_STATIC_LINK_CRT "Build Logtail by linking CRT statically" OFF)

编译时可以通过环境变量打开开关
ENABLE_COMPATIBLE_MODE=ON
ENABLE_STATIC_LINK_CRT=ON

**Reply by yanggelinux** [reply]
logtail 从1.2 升级到 1.3 版本之后,ilogtail-build-linux镜像也更新到最新版本后，这个方式不生效
